### ISSUE TITLE
feat(transforms): volume-preserving GIN coupling (Sorrenson 2020)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#68]
## Problem / Request
Add `GINCoupling`, the volume-preserving variant of affine coupling introduced in Sorrenson et al. 2020 ("Disentanglement by Nonlinear ICA with GIN"). Identical to affine coupling except the per-dim log-scales are centred so their sum is exactly zero — giving `log_det = 0` and a volume-preserving (but still expressive) layer.

## User Story
> As a researcher studying nonlinear ICA / disentanglement, I want a volume-preserving coupling block that keeps `log_det = 0` exactly, so that I can replicate GIN and downstream disentanglement-flow experiments without reimplementing from FrEIA.

## Motivation
- **Disentanglement.** GIN preserves the volume element; sufficient for identifiable nonlinear ICA under Sorrenson's theorem.
- **Numerical stability.** Zero logdet removes a noisy gradient term in the ELBO / NLL.
- **Cheap to add.** Mechanically a ~10-line modification of `AffineCoupling` — the only change is a centering op on `log_scale`.
- **Missing from gauss_flows.** No volume-preserving coupling currently.

## Proposed API
```python
from flowjax.bijections import AbstractBijection

class GINCoupling(AbstractBijection):
    """Volume-preserving affine coupling (Sorrenson et al. 2020).

    Identical to `AffineCoupling` except per-dim log-scales are centred so
    their sum is 0, yielding exactly log_det = 0.

    Args:
        key: PRNG key for the conditioner MLP.
        shape: Input shape (d,).
        cond_shape: Optional shape of the conditioning input.
        nn_width / nn_depth: Conditioner MLP config (matches AffineCoupling).
        mask: Optional Boolean mask selecting transformed dims.
    """
```

## Design Snapshot
```python
# src/gauss_flows/_src/transforms/bijections/coupling/gin_coupling.py
import jax.numpy as jnp

def _center_scales(log_scale):
    # Enforce Σ_i log_scale_i == 0 → det(diag(exp(log_scale))) == 1
    return log_scale - jnp.mean(log_scale, axis=-1, keepdims=True)

# Inside transform_and_log_det:
#   s, t = conditioner(x_cond)
#   s_centered = _center_scales(s)       # GIN
#   y_transform = x_transform * jnp.exp(s_centered) + t
#   log_det = 0.0                        # exact, not ε-close
```

## Mathematical Notes
```text
Standard affine coupling: y_A = x_A * exp(s) + t,              log|det J| = Σ_i s_i
GIN:                      y_A = x_A * exp(s - mean(s)) + t,    log|det J| = 0

This is a strict restriction of affine coupling to SL(d, R). Stacked, it is
still a universal approximator of volume-preserving diffeomorphisms.
```

## References & Existing Code
- Paper: Sorrenson, Rother, Köthe (2020). *Disentanglement by Nonlinear ICA with General Incompressible-flow Networks (GIN)*. ICLR 2020. arXiv:2001.04872
- Reference impl: `vislearn/FrEIA/FrEIA/modules/coupling_layers.py::GINCouplingBlock`
- gauss_flows analogue: `src/gauss_flows/_src/transforms/bijections/coupling/affine.py::AffineCoupling`

## Implementation Steps
- [ ] Add `GINCoupling` in `src/gauss_flows/_src/transforms/bijections/coupling/gin_coupling.py`
- [ ] Re-export from `src/gauss_flows/_src/transforms/bijections/coupling/__init__.py`
- [ ] Re-export from `src/gauss_flows/_src/transforms/bijections/__init__.py`, `src/gauss_flows/_src/transforms/__init__.py`, and `src/gauss_flows/__init__.py`
- [ ] Add `__all__` entries everywhere it's re-exported

## Definition of Done
- [ ] Code lands at the intended path
- [ ] Public API exported via `src/gauss_flows/__init__.py`
- [ ] Tests pass: `make test`
- [ ] Lint + typecheck pass: `make lint && make typecheck`
- [ ] Google-style docstring with citation

## Testing
- [ ] Unit test: `inverse(transform(x)) == x` to machine precision
- [ ] Unit test: `log_det == 0` exactly (`jnp.array_equal`, not `allclose`) for random inputs
- [ ] Unit test: volume preservation under composition — stack of 4 GIN layers still yields `log_det == 0`
- [ ] Regression test: scale-centering matches manual computation on a 3-D handcrafted example

## Documentation
- [ ] API reference entry

## Relationships
- Related: #65, #66, #67 (continuous-time neural flows), FrEIA port series
